components/RefinementList.js

```diff
@@ -9,10 +9,12 @@ class RefinementList extends React.Component {
 
   _generateFacetItem(facetValue) {
     var hasChildren = facetValue.data && facetValue.data.length > 0;
+    var subList = hasChildren && <RefinementList {...this.props} facetValues={facetValue.data} />;
+    var data = facetValue;
 
-    var subList = hasChildren ?
-      <RefinementList {...this.props} facetValues={facetValue.data} /> :
-      null;
```

Comment:
I think you are reffering to that article https://facebook.github.io/react/blog/2014/07/17/react-v0.11.html#rendering-to-null This doesn't apply here. It's only for the returned value of the render method that this apply.